packages/cli/src/databases/repositories/executionData.repository.ts

```diff
@@ -1,13 +1,33 @@
 import { Service } from 'typedi';
+import type { EntityManager } from '@n8n/typeorm';
+import type { IWorkflowBase } from 'n8n-workflow';
 import { DataSource, In, Repository } from '@n8n/typeorm';
 import { ExecutionData } from '../entities/ExecutionData';
 
+export interface CreateExecutionDataOpts extends Pick<ExecutionData, 'data' | 'executionId'> {
+	workflowData: Pick<IWorkflowBase, 'connections' | 'nodes' | 'name' | 'settings' | 'id'>;
+}
+
 @Service()
 export class ExecutionDataRepository extends Repository<ExecutionData> {
 	constructor(dataSource: DataSource) {
 		super(ExecutionData, dataSource.manager);
 	}
 
+	async createExecutionDataForExecution(
+		executionData: CreateExecutionDataOpts,
+		trx?: EntityManager,
```

Comment:
isn't `trx` always passed in so far? maybe we can make this optional when we actually need to?